_source/_posts/2020-06-29-aspnet-core-mongodb.md

```diff
@@ -0,0 +1,513 @@
+        <h2>Submit an Idea</h2>
+        @Html.TextAreaFor(m=>m.Content)
+        <input type="submit" name="Create" />
+    </fieldset>
+</form>
+```
+
+**Edit.cshtml**
+```html
+@model OktaNetCoreMvcMongoExample.Models.Submission
+
+<form asp-action="Edit">
+    <fieldset class="idea-form">
+        @Html.HiddenFor(m => m.Created)
+        @Html.HiddenFor(m=>m.UserId)
+        @Html.HiddenFor(m=>m.UserName)
+        @Html.LabelFor(m => m.Content)
+        @Html.TextAreaFor(m => m.Content)
+        <input type="submit" name="Update" />
+    </fieldset>
+</form>
+```
+
+The only things really remarkable here are the hidden fields for the `Created`, `UserId`, and `UserName` properties in the edit form. These will ensure the values are passed back without exposing them to edits.
+
+I added a little CSS magic to make the views look a little more presentable. In the `wwwroot/css/site.css` file at the bottom, I added:
+
+```css
+/* Idea specific styles */
+
+.idea {
+  border: solid 1px #DDD;
+  padding: .5rem;
+  margin-bottom: .5rem;
+}
+
+.idea > span {
+  font-size: .75rem;
+  color: #AAA;
+  font-style: italic;
+  padding: .5rem;
+}
+
+.idea > div {
+  padding: .5rem;
+}
+
+a.create {
+  display: block;
+  width: 100%;
+  text-align: right;
+}
+
+.idea-form > * {
+  display: block;
+}
+
+.idea-form > textarea {
+  height: 5rem;
+  width: 100%;
+}
+```
+
+## Add Authentication to the .NET Core Application
+
+Lastly, you'll need to add authentication using Okta so that the application can enforce the rule that only users who created a submission can edit or delete those submissions. Start by creating an application in Okta—go to your Okta dashboard, click on **Applications** in the top menu and then click the green **Add Application** button. Choose **Web** and click **Next**.
+
+In the **Application Settings**, name the application "SuggestMe" then change the **Base URIs** to `http://localhost:5000/`, the **Login redirect URIs** to `http://localhost:5000/signin-oidc` and the **Logout redirect URIs** to `http://localhost:5000/signout-callback-oidc`. All the other default values should be fine.
+
+{% img blog/aspnet-core-mongodb/okta-app-settings.png alt:"Okta App Settings" width:"800" %}{: .center-image }
+
+When you click **Done**, you should be directed to the application settings page. On the **General** tab at the bottom, you'll see a box with a **ClientId** and **ClientSecret**. Copy those to a new section in `appsettings.json` below the database settings you added earlier:
+
+```json
+"Okta": {
+    "ClientId": "{yourClientId}",
+    "ClientSecret": "{yourClientSecret}",
+    "Domain": "{yourOktaDomain}",
+    "PostLogoutRedirectUri": "http://localhost:5000"
+},
+```
+
+You can find your Okta domain URL on the top right of your Okta dashboard.
+
+{% img blog/aspnet-core-mongodb/okta-org-url.png alt:"Okta Org URL" width:"800" %}{: .center-image }
+
+Next, add a file called `AccountController.cs` to the `Controllers` folder like so:
+
+```cs
```

Comment:
This code block indents by 2 spaces while the rest have 4. I'd make it consistent.